actions/generate-schedule/users/usersource.go

```diff
@@ -0,0 +1,52 @@
+package users
+
+import (
+	"container/list"
+	"fmt"
+	"sort"
+)
+
+// Source represents a source of usernames for rotation
+type Source interface {
+	StartAfter(user string) error
+	NextUser() string
+}
+
+// StaticSource is a base implementation of static list of usernames.
+type StaticSource struct {
+	users    *list.List
```

Comment:
Looks like a [Ring](https://golang.org/pkg/container/ring/) might be a better data structure for what we're doing?

Reply:
💍 🎉 
This is the best thing I've learned all week!

Reply:
Nice - makes it a bit smaller.